wimm.Secundatives.UnitTests/Extensions/ResultMapping_Test.cs

```diff
@@ -0,0 +1,186 @@
+﻿using System.Threading.Tasks;
+using Xunit;
+
+namespace wimm.Secundatives.UnitTests.Extensions
+{
+    public class ResultMapping_Test
+    {
+        [Fact]
+        public void MapResult_IsSuccessAndFuncReturnsValue_ReturnsResultOfFunc()
+        {
+            var underTest = ConstructWith("doot");
+            var res = underTest.Map(x => "deet");
+
+            Assert.Equal("deet", res.Value);
+        }
+
+        [Fact]
+        public void MapResult_IsErrorAndFuncReturnsValue_ReturnsError()
+        {
+            var underTest = new Result<string, TestError>(TestError.Sadness);
+
+            var res = underTest.Map(x => "deet");
+
+            Assert.True(res.IsError);
+            Assert.Equal(TestError.Sadness, res.Error);
+
+        }
+
+        [Fact]
+        public async Task MapResult_IsSuccessAndFuncReturnsTask_ReturnsResultOfFunc()
+        {
+            var underTest = ConstructWith("doot");
+            var res = await underTest.Map(async x => await Task.FromResult("skeleton"));
```

Comment:
not that it matters, but is there any reason in the tests to do 
```
var res = await underTest.Map(async x => await Task.FromResult("skeleton"));
```
over just
```
var res = await underTest.Map(x => Task.FromResult("skeleton"));
```
?

Reply:
I was super tired